test/integration/sidecar/sidecar_stack.go

```diff
@@ -0,0 +1,448 @@
+package sidecar
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	appmesh "github.com/aws/aws-app-mesh-controller-for-k8s/apis/appmesh/v1beta2"
+	"github.com/aws/aws-app-mesh-controller-for-k8s/test/framework"
+	"github.com/aws/aws-sdk-go/aws"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+const (
+	defaultFrontendImage = "public.ecr.aws/b7m0w2t6/color-fe-app:2.0.3"
+	defaultBackendImage  = "public.ecr.aws/b7m0w2t6/color-be-app:2.0.2"
+)
+
+type SidecarStack struct {
+	appContainerPort int
+	color            string
+	k8client         *kubernetes.Clientset
+	testName         string
+
+	mesh      *appmesh.Mesh
+	namespace *corev1.Namespace
+}
+
+func newSidecarStack(name string, kubecfg string) (*SidecarStack, error) {
+	cfg, err := clientcmd.BuildConfigFromFlags("", kubecfg)
+	if err != nil {
+		return nil, err
+	}
+
+	k8client, err := kubernetes.NewForConfig(cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	return &SidecarStack{
+		appContainerPort: 8080,
+		color:            "blue",
+		testName:         name,
+		k8client:         k8client,
+	}, nil
+}
+
+func (s *SidecarStack) createMeshAndNamespace(ctx context.Context, f *framework.Framework) {
+	By("create Mesh", func() {
+		mesh := &appmesh.Mesh{
+			ObjectMeta: metav1.ObjectMeta{Name: s.testName},
+			Spec: appmesh.MeshSpec{
+				NamespaceSelector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"mesh": s.testName,
+					},
+				},
+				EgressFilter: &appmesh.EgressFilter{
+					Type: appmesh.EgressFilterTypeAllowAll,
+				},
+			},
+		}
+
+		err := f.K8sClient.Create(ctx, mesh)
+		Expect(err).NotTo(HaveOccurred())
+
+		_, err = f.MeshManager.WaitUntilMeshActive(ctx, mesh)
+		Expect(err).NotTo(HaveOccurred())
+
+		s.mesh = mesh
+	})
+
+	By("create Namespace", func() {
+		namespace := &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: s.testName,
+				Labels: map[string]string{
+					"appmesh.k8s.aws/sidecarInjectorWebhook": "enabled",
+					"mesh":                                   s.mesh.Name,
+				},
+			},
+		}
+
+		err := f.K8sClient.Create(ctx, namespace)
+		Expect(err).NotTo(HaveOccurred())
+
+		s.namespace = namespace
+	})
+
+	By("create Role", func() {
+		role := &rbacv1.Role{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "default",
+				Namespace: s.testName,
+			},
+			Rules: []rbacv1.PolicyRule{
+				{
+					APIGroups: []string{""},
+					Resources: []string{"pods"},
+					Verbs:     []string{"list", "update"},
```

Comment:
we need this role and the subsequent role binding to get a list of the pods and update the pod annotation from within the app container